src/reducers/columnStateHelper.js

```diff
@@ -47,79 +42,180 @@ function initialize(state, props, oldProps) {
-  const fixedColumnsCount = fixedColumns.length;
-  const scrollableColumnsCount = scrollableColumns.length;
-
-  const noScrollableColumns = scrollableColumnsCount === 0;
-  const scrollToUnchanged = scrollToColumn === oldScrollToColumn;
-  const selectedColumnFixed = scrollToColumn < fixedColumnsCount;
-  const selectedColumnFixedRight = scrollToColumn >= fixedColumnsCount + scrollableColumnsCount;
-  if (scrollToUnchanged || selectedColumnFixed || selectedColumnFixedRight || noScrollableColumns) {
-    return scrollX;
+
+  const { scrollX, columnOffsetIntervalTree } = state;
+
+  const selectedColumnFixed = scrollToColumn < fixedColumns.length;
+  const selectedColumnFixedRight = scrollToColumn >= fixedColumns.length + scrollableColumns.length;
+
+  // if target column is fixed, then don't do anything
+  if (selectedColumnFixed || selectedColumnFixedRight) {
+    return {
+      firstIndex: state.firstColumnIndex,
+      firstOffset: state.firstColumnOffset,
+      lastIndex: undefined,
+      changed: false,
+    }
   }
 
-  // Convert column index (0 indexed) to scrollable index (0 indexed)
-  // and clamp to max scrollable index
-  const clampedColumnIndex = Math.min(scrollToColumn - fixedColumnsCount,
-    scrollableColumns.length - 1);
+  // convert to scrollable column index where 0 denotes first scrollable column
+  const clampedColumnIndex = Math.min(scrollToColumn - fixedColumns.length, scrollableColumns.length - 1);
+  const columnBeginX = columnOffsetIntervalTree.sumUntil(clampedColumnIndex);
+  const columnEndX = columnBeginX + updateColumnWidth(state, clampedColumnIndex);
 
-  // Compute the width of all columns to the left of the column
-  let previousWidth = 0;
-  for (let columnIdx = 0; columnIdx < clampedColumnIndex; ++columnIdx) {
-    previousWidth += scrollableColumns[columnIdx].width;
+  let firstIndex = clampedColumnIndex;
+  let lastIndex = undefined;
+  if (columnBeginX < scrollX) {
+    // If column starts before the viewport, set as the first column in the viewport
+    // Uses defaults (from above)
+  } else if (scrollX < columnEndX - availableScrollWidth) {
+    // If after the viewport, set as the last column in the viewport
+    firstIndex = undefined;
+    lastIndex = clampedColumnIndex;
+  } else {
+    // If already in the viewport, do nothing.
+    return {
+      firstIndex: state.firstColumnIndex,
+      firstOffset: state.firstColumnOffset,
+      lastIndex: undefined,
+      changed: false,
+    };
   }
 
-  // Get width of specified column
-  const selectedColumnWidth = scrollableColumns[clampedColumnIndex].width;
+  return {
+    firstIndex,
+    firstOffset: 0,
+    lastIndex,
+    changed: true,
+  };
+}
+
+/**
+ * Scroll to a specific position in the grid
+ *
+ * @param {!Object} state
+ * @param {number} scrollX
+ * @return {{
+ *   firstIndex: number,
+ *   firstOffset: number,
+ *   lastIndex: number,
+ *   changed: number
+ * }}
+ */
+function scrollToPos(state, scrollX) {
```

Comment:
similar to scrollAnchor's [scrollTo](https://github.com/schrodinger/fixed-data-table-2/blob/v1.0-beta/src/reducers/scrollAnchor.js#L67)